src/scib_metrics/_lisi.py

```diff
@@ -0,0 +1,121 @@
+from typing import Tuple
+
+import numpy as np
+from scipy.sparse import csr_matrix
+from sklearn.neighbors import NearestNeighbors
+from sklearn.utils import check_array
+
+from scib_metrics.utils import compute_simpson_index
+
+
+def _convert_knn_graph_to_idx(knn_graph: csr_matrix) -> Tuple[np.ndarray, np.ndarray]:
+    check_array(knn_graph, accept_sparse="csr")
+
+    n_neighbors = np.unique(knn_graph.nonzero()[0], return_counts=True)[1]
+    if len(np.unique(n_neighbors)) > 1:
+        raise ValueError("Each cell must have the same number of neighbors.")
+
+    n_neighbors = int(np.unique(n_neighbors)[0])
+
+    nn_obj = NearestNeighbors(n_neighbors=n_neighbors, metric="precomputed").fit(knn_graph)
+    return nn_obj.kneighbors(knn_graph)
+
+
+def lisi_knn(knn_graph: csr_matrix, labels: np.ndarray, perplexity: float = None) -> np.ndarray:
+    """Compute the local inverse simpson index (LISI) for each cell.
+
+    Code inspired by:
+    https://github.com/theislab/scib/blob/e578d84063adf4853ed087500bd3d67078e53337/scib/metrics/lisi.py#L586
```

Comment:
Instead of this can you add a bib entry for the harmony paper like this?

https://github.com/YosefLab/scib-metrics/blob/a258e0b8bea44549d0225c0e1181b13c556049e2/src/scib_metrics/_silhouette.py#L8

and then the scib paper in the rescaled ones

Reply:
gonna need to commit the other one first for this